src/coreclr/vm/class.cpp

```diff
@@ -1222,6 +1222,17 @@ void ClassLoader::LoadExactParents(MethodTable* pMT)
     // We can now mark this type as having exact parents
     pMT->SetHasExactParent();
 
+#ifdef TARGET_RISCV64
+    if (pMT->HasLayout() &&
+        pMT->GetVerifierCorElementType() == ELEMENT_TYPE_VALUETYPE &&
+        pMT->GetNumInstanceFieldBytes() <= ENREGISTERED_RETURNTYPE_MAXSIZE)
+    {
+        // To calculate how a small struct is enregistered RISC-V needs native layout info.
+        // Initialize now so ArgIterator doesn't trigger undue class loads during GC stack walks.
+        pMT->EnsureNativeLayoutInfoInitialized();
```

Comment:
What is it that makes this necessary on RISC-V but not on anything else?

Reply:
RISC-V has more complicated calling convention than anything else. More information needs to be computed for RISC-V to interpret the calling convention. This information needs to be computed upfront - we are not able to compute it at the point where the calling convention is interpreted.

Reply:
Hmm ... I thought I was reasonably familiar with the ABI, and that it is pretty similar to A64. The only thing I'd regard as a little funky is structs with one int and one fp value being passed in the next int and the next fp register, if available. Otherwise ... big things are passed by reference, return values are passed the same as the first argument, big return values pass a pointer to a buf as the first int argument, bumping the rest along. All pretty standard stuff.

Is there some particular corner case that makes it more complex?

Reply:
Yes, GetNativeLayoutInfo() in particular is causing problems as it can trigger a full class load (in CoreLibBinder::GetClass) here:
https://github.com/dotnet/runtime/blob/0a5418d7e7ae7f7653a32004c808881af5275469/src/coreclr/vm/classlayoutinfo.cpp#L1027-L1041
I've been looking at calling EnsureEnsureNativeLayoutInfoInitialized() before SetHasExactParent() but then CoreLibBinder::GetClass(CLASS__INT128) triggers EnsureEnsureNativeLayoutInfoInitialized() for it which then tries to load Int128 (at this point not fully loaded yet), which fails on a deadlock check:
https://github.com/dotnet/runtime/blob/0a5418d7e7ae7f7653a32004c808881af5275469/src/coreclr/vm/clsload.cpp#L3181-L3185
I'll come back to it to figure out how to side-step this issue. It's a bit unfortunate EEClassNativeLayoutInfo::CollectNativeLayoutFieldMetadataThrowing() is doing a full class load, MethodTable::HasSameTypeDefAs() needs just the RID which I believe we get much earlier in the load sequence and the module which is a known global in CoreLibBinder.

Reply:
The RID formula is not fixed. The RID changes from build to build.

If all you need is the RID, you can load the Vector type explicitly upfront here: https://github.com/dotnet/runtime/blob/18b08169b979084fd038cb2d848568ceb4049f01/src/coreclr/vm/appdomain.cpp#L1380

Reply:
> If all you need is the RID, you can load the Vector type explicitly upfront

Preloading classes for comparison gets rid of the direct problem in EEClassNativeLayoutInfo::CollectNativeLayoutFieldMetadataThrowing, thank you. However, I'm still hitting recursive class loads while constructing MarshalInfo here:
https://github.com/dotnet/runtime/blob/db8f9eb77484629855fcbf91e70016c3c868290b/src/coreclr/vm/mlinfo.cpp#L1735-L1740

Reply:
What is the stack trace and the type that you are hitting this for?

Reply:
> What is the stack trace and the type that you are hitting this for?

<details>
  <summary>Stack trace, some line numbers are slightly off because I inserted deoptimizing attributes</summary>

```c++
#0  ClassLoader::ThrowTypeLoadException (pKey=0x400180af28, resIDWhy=2148734242) at /runtime/src/coreclr/vm/clsload.cpp:737
#1  0x0000004001ed92aa in ClassLoader::LoadTypeHandleForTypeKey_Body (this=0x40001a97a0, pTypeKey=0x400180af28, typeHnd=..., targetLevel=CLASS_LOAD_EXACTPARENTS)
    at /runtime/src/coreclr/vm/clsload.cpp:3190
#2  0x0000004001ed4994 in ClassLoader::LoadTypeHandleForTypeKey (this=0x40001a97a0, pTypeKey=0x400180af28, typeHnd=..., targetLevel=CLASS_LOAD_EXACTPARENTS, pInstContext=0x0)
    at /runtime/src/coreclr/vm/clsload.cpp:3011
#3  0x0000004001ed60d0 in ClassLoader::LoadTypeDefThrowing (pModule=0x3f833bf130, typeDef=33555137, fNotFoundAction=ClassLoader::ThrowIfNotFound,
    fUninstantiated=ClassLoader::FailIfUninstDefOrRef, tokenNotToLoad=0, level=CLASS_LOAD_EXACTPARENTS, pTargetInstantiation=0x0) at /runtime/src/coreclr/vm/clsload.cpp:1998
#4  0x0000004001ed6e1e in ClassLoader::LoadTypeDefOrRefThrowing (pModule=0x3f833bf130, typeDefOrRef=33555137, fNotFoundAction=ClassLoader::ThrowIfNotFound,
    fUninstantiated=ClassLoader::FailIfUninstDefOrRef, tokenNotToLoad=0, level=CLASS_LOAD_EXACTPARENTS) at /runtime/src/coreclr/vm/clsload.cpp:2167
#5  0x0000004001fa8dda in SigPointer::GetTypeHandleThrowing (this=0x400180c500, pModule=0x3f833bf130, pTypeContext=0x400180c980, fLoadTypes=ClassLoader::LoadTypes,
    level=CLASS_LOAD_EXACTPARENTS, dropGenericArgumentLevel=0, pSubst=0x0, pZapSigContext=0x0, pMTInterfaceMapOwner=0x0, pRecursiveFieldGenericHandling=0x0)
    at /runtime/src/coreclr/vm/siginfo.cpp:1641
#6  0x0000004001fa88b2 in SigPointer::GetTypeHandleThrowing (this=0x400180c9c0, pModule=0x3f833bf130, pTypeContext=0x400180c980, fLoadTypes=ClassLoader::LoadTypes,
    level=CLASS_LOAD_EXACTPARENTS, dropGenericArgumentLevel=0, pSubst=0x0, pZapSigContext=0x0, pMTInterfaceMapOwner=0x0, pRecursiveFieldGenericHandling=0x0)
    at /runtime/src/coreclr/vm/siginfo.cpp:1505
#7  0x0000004001f0ea02 in MetaSig::GetLastTypeHandleThrowing (this=0x400180c978, fLoadTypes=ClassLoader::LoadTypes, level=<optimized out>, dropGenericArgumentLevel=<optimized out>)
    at /runtime/src/coreclr/vm/siginfo.hpp:917
#8  FieldDesc::GetFieldTypeHandleThrowing (this=<optimized out>, level=29345424, dropGenericArgumentLevel=2048) at /runtime/src/coreclr/vm/field.cpp:166
#9  0x0000004001f763e8 in MethodTable::DoFullyLoad (this=0x3f8432e568, pVisited=0x400180ceb8, level=CLASS_DEPENDENCIES_LOADED, pPending=0x0, pfBailed=0x400180cedc, pInstContext=0x0)
    at /runtime/src/coreclr/vm/methodtable.cpp:5197
#10 0x0000004001f7631e in MethodTable::DoFullyLoad (this=0x3f8432edb0, pVisited=0x400180d188, level=CLASS_DEPENDENCIES_LOADED, pPending=0x0, pfBailed=0x400180d1ac, pInstContext=0x0)
    at /runtime/src/coreclr/vm/methodtable.cpp:5176
#11 0x0000004001f7625a in MethodTable::DoFullyLoad (this=0x3f843b9e90, pVisited=0x0, level=CLASS_DEPENDENCIES_LOADED, pPending=0x0, pfBailed=0x400180d2dc, pInstContext=0x0)
    at /runtime/src/coreclr/vm/methodtable.cpp:5157
#12 0x0000004001ed9cce in PushFinalLevels (typeHnd=..., targetLevel=CLASS_LOADED, pInstContext=0x0) at /runtime/src/coreclr/vm/clsload.cpp:2941
#13 0x0000004001ed49f2 in ClassLoader::LoadTypeHandleForTypeKey (this=0x40001a97a0, pTypeKey=0x400180d3b8, typeHnd=..., targetLevel=CLASS_LOADED, pInstContext=0x0)
    at /runtime/src/coreclr/vm/clsload.cpp:3018
#14 0x0000004001ed60d0 in ClassLoader::LoadTypeDefThrowing (pModule=0x3f833bf130, typeDef=33554907, fNotFoundAction=ClassLoader::ThrowIfNotFound,
    fUninstantiated=ClassLoader::FailIfUninstDefOrRef, tokenNotToLoad=0, level=CLASS_LOADED, pTargetInstantiation=0x0) at /runtime/src/coreclr/vm/clsload.cpp:1998
#15 0x0000004001ed6e1e in ClassLoader::LoadTypeDefOrRefThrowing (pModule=0x3f833bf130, typeDefOrRef=33554907, fNotFoundAction=ClassLoader::ThrowIfNotFound,
    fUninstantiated=ClassLoader::FailIfUninstDefOrRef, tokenNotToLoad=0, level=CLASS_LOADED) at /runtime/src/coreclr/vm/clsload.cpp:2167
#16 0x0000004001fa8dda in SigPointer::GetTypeHandleThrowing (this=0x400180ef50, pModule=0x3f833bf130, pTypeContext=0x400180f2e8, fLoadTypes=ClassLoader::LoadTypes,
    level=CLASS_LOADED, dropGenericArgumentLevel=0, pSubst=0x0, pZapSigContext=0x0, pMTInterfaceMapOwner=0x0, pRecursiveFieldGenericHandling=0x0)
    at /runtime/src/coreclr/vm/siginfo.cpp:1641
#17 0x00000040020c34c2 in MarshalInfo::MarshalInfo (this=0x400180ef60, pModule=0x3f833bf130, sig=..., pTypeContext=0x400180f2e8, token=67112859,
    ms=MarshalInfo::MARSHAL_SCENARIO_FIELD, nlType=nltAnsi, nlFlags=nlfNone, isParam=0, paramidx=0, numArgs=0, BestFit=0, ThrowOnUnmappableChar=0, fEmitsIL=0, pMD=0x0,
    fLoadCustomMarshal=0, pDebugName=0x6091d48ff9 "ContainingImportSection", pDebugClassName=0x6091d552be "ImportThunkKey", argidx=4294967295)
    at /runtime/src/coreclr/vm/mlinfo.cpp:1742
#18 0x0000004002061abe in ParseNativeType (pModule=0x3f833bf130, sig=..., pFD=0x3f8432d1c8, flags=ParseNativeTypeFlags::IsAnsi, pNFD=0x608c029d40, pTypeContext=0x400180f2e8,
    szNamespace=0x6091d32afc "", szClassName=0x6091d552be "ImportThunkKey", szFieldName=0x6091d48ff9 "ContainingImportSection") at /runtime/src/coreclr/vm/fieldmarshaler.cpp:56
#19 0x0000004001ed1bae in (anonymous namespace)::ParseFieldNativeTypes (pInternalImport=0x40001b1e70, cl=33555137, fieldDescs=..., pModule=0x3f833bf130,
    nativeTypeFlags=ParseNativeTypeFlags::IsAnsi, pTypeContext=0x400180f2e8, pFieldInfoArrayOut=0x608c029d28, szNamespace=0x6091d32afc "", szName=0x6091d552be "ImportThunkKey")
    at /runtime/src/coreclr/vm/classlayoutinfo.cpp:442
#20 0x0000004001ed0d88 in EEClassNativeLayoutInfo::CollectNativeLayoutFieldMetadataThrowing (pMT=0x3f8432d430) at /runtime/src/coreclr/vm/classlayoutinfo.cpp:923
#21 0x0000004001ed05bc in EEClassNativeLayoutInfo::InitializeNativeLayoutFieldMetadataThrowing (pMT=0x3f8432d430) at /runtime/src/coreclr/vm/classlayoutinfo.cpp:844
#22 0x0000004001f7f4b0 in MethodTable::EnsureNativeLayoutInfoInitialized (this=0x3f8432d430) at /runtime/src/coreclr/vm/methodtable.cpp:9280
#23 0x0000004001ec7986 in ClassLoader::LoadExactParents (pMT=0x3f8432d430) at /runtime/src/coreclr/vm/class.cpp:1231
#24 0x0000004001ed7ec0 in ClassLoader::DoIncrementalLoad (pTypeKey=0x400180fc58, typeHnd=..., currentLevel=CLASS_LOAD_APPROXPARENTS) at /runtime/src/coreclr/vm/clsload.cpp:2634
#25 0x0000004001ed96a6 in ClassLoader::LoadTypeHandleForTypeKey_Body (this=0x40001a97a0, pTypeKey=0x400180fc58, typeHnd=..., targetLevel=CLASS_LOAD_EXACTPARENTS)
    at /runtime/src/coreclr/vm/clsload.cpp:3283
#26 0x0000004001ed4994 in ClassLoader::LoadTypeHandleForTypeKey (this=0x40001a97a0, pTypeKey=0x400180fc58, typeHnd=..., targetLevel=CLASS_LOAD_EXACTPARENTS, pInstContext=0x0)
    at /runtime/src/coreclr/vm/clsload.cpp:3011
#27 0x0000004001ed60d0 in ClassLoader::LoadTypeDefThrowing (pModule=0x3f833bf130, typeDef=33555137, fNotFoundAction=ClassLoader::ThrowIfNotFound,
    fUninstantiated=ClassLoader::FailIfUninstDefOrRef, tokenNotToLoad=0, level=CLASS_LOAD_EXACTPARENTS, pTargetInstantiation=0x0) at /runtime/src/coreclr/vm/clsload.cpp:1998
#28 0x0000004001ed6e1e in ClassLoader::LoadTypeDefOrRefThrowing (pModule=0x3f833bf130, typeDefOrRef=33555137, fNotFoundAction=ClassLoader::ThrowIfNotFound,
    fUninstantiated=ClassLoader::FailIfUninstDefOrRef, tokenNotToLoad=0, level=CLASS_LOAD_EXACTPARENTS) at /runtime/src/coreclr/vm/clsload.cpp:2167
#29 0x0000004001fa8dda in SigPointer::GetTypeHandleThrowing (this=0x4001811230, pModule=0x3f833bf130, pTypeContext=0x40018116b0, fLoadTypes=ClassLoader::LoadTypes,
    level=CLASS_LOAD_EXACTPARENTS, dropGenericArgumentLevel=0, pSubst=0x0, pZapSigContext=0x0, pMTInterfaceMapOwner=0x0, pRecursiveFieldGenericHandling=0x0)
    at /runtime/src/coreclr/vm/siginfo.cpp:1641
#30 0x0000004001fa88b2 in SigPointer::GetTypeHandleThrowing (this=0x40018116f0, pModule=0x3f833bf130, pTypeContext=0x40018116b0, fLoadTypes=ClassLoader::LoadTypes,
    level=CLASS_LOAD_EXACTPARENTS, dropGenericArgumentLevel=0, pSubst=0x0, pZapSigContext=0x0, pMTInterfaceMapOwner=0x0, pRecursiveFieldGenericHandling=0x0)
    at /runtime/src/coreclr/vm/siginfo.cpp:1505
#31 0x0000004001f0ea02 in MetaSig::GetLastTypeHandleThrowing (this=0x40018116a8, fLoadTypes=ClassLoader::LoadTypes, level=<optimized out>, dropGenericArgumentLevel=<optimized out>)
    at /runtime/src/coreclr/vm/siginfo.hpp:917
#32 FieldDesc::GetFieldTypeHandleThrowing (this=<optimized out>, level=29345424, dropGenericArgumentLevel=2048) at /runtime/src/coreclr/vm/field.cpp:166
#33 0x0000004001f763e8 in MethodTable::DoFullyLoad (this=0x3f8432e568, pVisited=0x4001811be8, level=CLASS_DEPENDENCIES_LOADED, pPending=0x0, pfBailed=0x4001811c0c, pInstContext=0x0)
    at /runtime/src/coreclr/vm/methodtable.cpp:5197
#34 0x0000004001f7631e in MethodTable::DoFullyLoad (this=0x3f8432edb0, pVisited=0x4001811eb8, level=CLASS_DEPENDENCIES_LOADED, pPending=0x0, pfBailed=0x4001811edc, pInstContext=0x0)
--Type <RET> for more, q to quit, c to continue without paging--c
    at /runtime/src/coreclr/vm/methodtable.cpp:5176
#35 0x0000004001f7625a in MethodTable::DoFullyLoad (this=0x3f8432e968, pVisited=0x4001812188, level=CLASS_DEPENDENCIES_LOADED, pPending=0x0, pfBailed=0x40018121ac, pInstContext=0x0) at /runtime/src/coreclr/vm/methodtable.cpp:5157
#36 0x0000004001f761d0 in MethodTable::DoFullyLoad (this=0x3f84327c00, pVisited=0x4001812458, level=CLASS_DEPENDENCIES_LOADED, pPending=0x0, pfBailed=0x400181247c, pInstContext=0x0) at /runtime/src/coreclr/vm/methodtable.cpp:5138
#37 0x0000004001f761d0 in MethodTable::DoFullyLoad (this=0x3f843280d0, pVisited=0x4001812728, level=CLASS_DEPENDENCIES_LOADED, pPending=0x0, pfBailed=0x400181274c, pInstContext=0x0) at /runtime/src/coreclr/vm/methodtable.cpp:5138
#38 0x0000004001f761d0 in MethodTable::DoFullyLoad (this=0x3f84328ac8, pVisited=0x0, level=CLASS_DEPENDENCIES_LOADED, pPending=0x0, pfBailed=0x400181287c, pInstContext=0x0) at /runtime/src/coreclr/vm/methodtable.cpp:5138
#39 0x0000004001ed9cce in PushFinalLevels (typeHnd=..., targetLevel=CLASS_LOADED, pInstContext=0x0) at /runtime/src/coreclr/vm/clsload.cpp:2941
#40 0x0000004001ed49f2 in ClassLoader::LoadTypeHandleForTypeKey (this=0x40001a97a0, pTypeKey=0x4001812958, typeHnd=..., targetLevel=CLASS_LOADED, pInstContext=0x0) at /runtime/src/coreclr/vm/clsload.cpp:3018
#41 0x0000004001ed60d0 in ClassLoader::LoadTypeDefThrowing (pModule=0x3f833bf130, typeDef=33554886, fNotFoundAction=ClassLoader::ThrowIfNotFound, fUninstantiated=ClassLoader::FailIfUninstDefOrRef, tokenNotToLoad=0, level=CLASS_LOADED, pTargetInstantiation=0x0) at /runtime/src/coreclr/vm/clsload.cpp:1998
#42 0x0000004001ed6e1e in ClassLoader::LoadTypeDefOrRefThrowing (pModule=0x3f833bf130, typeDefOrRef=33554886, fNotFoundAction=ClassLoader::ThrowIfNotFound, fUninstantiated=ClassLoader::FailIfUninstDefOrRef, tokenNotToLoad=0, level=CLASS_LOADED) at /runtime/src/coreclr/vm/clsload.cpp:2167
#43 0x0000004001fa8dda in SigPointer::GetTypeHandleThrowing (this=0x40018137a0, pModule=0x3f833bf130, pTypeContext=0x4001813778, fLoadTypes=ClassLoader::LoadTypes, level=CLASS_LOADED, dropGenericArgumentLevel=0, pSubst=0x0, pZapSigContext=0x0, pMTInterfaceMapOwner=0x0, pRecursiveFieldGenericHandling=0x0) at /runtime/src/coreclr/vm/siginfo.cpp:1641
#44 0x0000004001f48a52 in CEEInfo::getArgClass (this=<optimized out>, sig=0x4001814420, args=<optimized out>) at /runtime/src/coreclr/vm/jitinterface.cpp:9693
#45 0x00000060890d1354 in Compiler::lvaInitTypeRef (this=0x400044cf48) at /runtime/src/coreclr/jit/lclvars.cpp:301
#46 0x0000006088fe425a in Compiler::compCompileHelper (this=0x400044cf48, classPtr=0x3f833bf130, compHnd=0x4001814528, methodInfo=0x4001814380, methodCodePtr=0x40018141a8, methodCodeSize=0x400181435c, compileFlags=0x40018141c8) at /runtime/src/coreclr/jit/compiler.cpp:6945
#47 0x0000006088fe310a in Compiler::compCompile(CORINFO_MODULE_STRUCT_*, void**, unsigned int*, JitFlags*)::$_0::operator()(Compiler::compCompile(CORINFO_MODULE_STRUCT_*, void**, unsigned int*, JitFlags*)::__JITParam*) const (this=<optimized out>, __JITpParam=<optimized out>) at /runtime/src/coreclr/jit/compiler.cpp:6355
#48 Compiler::compCompile (this=0x400044cf48, classPtr=0x3f833bf130, methodCodePtr=0x40018141a8, methodCodeSize=0x400181435c, compileFlags=0x40018141c8) at /runtime/src/coreclr/jit/compiler.cpp:6374
#49 0x0000006088fe5954 in jitNativeCode(CORINFO_METHOD_STRUCT_*, CORINFO_MODULE_STRUCT_*, ICorJitInfo*, CORINFO_METHOD_INFO*, void**, unsigned int*, JitFlags*, void*)::$_0::operator()(jitNativeCode(CORINFO_METHOD_STRUCT_*, CORINFO_MODULE_STRUCT_*, ICorJitInfo*, CORINFO_METHOD_INFO*, void**, unsigned int*, JitFlags*, void*)::__JITParam*) const::{lambda(jitNativeCode(CORINFO_METHOD_STRUCT_*, CORINFO_MODULE_STRUCT_*, ICorJitInfo*, CORINFO_METHOD_INFO*, void**, unsigned int*, JitFlags*, void*)::$_0::operator()(jitNativeCode(CORINFO_METHOD_STRUCT_*, CORINFO_MODULE_STRUCT_*, ICorJitInfo*, CORINFO_METHOD_INFO*, void**, unsigned int*, JitFlags*, void*)::__JITParam*) const::__JITParam*)#1}::operator()(jitNativeCode(CORINFO_METHOD_STRUCT_*, CORINFO_MODULE_STRUCT_*, ICorJitInfo*, CORINFO_METHOD_INFO*, void**, unsigned int*, JitFlags*, void*)::$_0::operator()(jitNativeCode(CORINFO_METHOD_STRUCT_*, CORINFO_MODULE_STRUCT_*, ICorJitInfo*, CORINFO_METHOD_INFO*, void**, unsigned int*, JitFlags*, void*)::__JITParam*) const::__JITParam*) const (this=<optimized out>, __JITpParam=<optimized out>) at /runtime/src/coreclr/jit/compiler.cpp:7864
#50 jitNativeCode(CORINFO_METHOD_STRUCT_*, CORINFO_MODULE_STRUCT_*, ICorJitInfo*, CORINFO_METHOD_INFO*, void**, unsigned int*, JitFlags*, void*)::$_0::operator()(jitNativeCode(CORINFO_METHOD_STRUCT_*, CORINFO_MODULE_STRUCT_*, ICorJitInfo*, CORINFO_METHOD_INFO*, void**, unsigned int*, JitFlags*, void*)::__JITParam*) const (__JITpParam=<optimized out>, this=<optimized out>) at /runtime/src/coreclr/jit/compiler.cpp:7889
#51 jitNativeCode (methodHnd=0x3f840de6b8, classPtr=0x3f833bf130, compHnd=0x4001814528, methodInfo=0x4001814380, methodCodePtr=0x40018141a8, methodCodeSize=0x400181435c, compileFlags=0x40018141c8, inlineInfoPtr=0x0) at /runtime/src/coreclr/jit/compiler.cpp:7891
#52 0x0000006088fee534 in CILJit::compileMethod (this=<optimized out>, compHnd=0x4001814528, methodInfo=0x4001814380, flags=<optimized out>, entryAddress=0x4001814ed0, nativeSizeOfCode=0x400181435c) at /runtime/src/coreclr/jit/ee_il_dll.cpp:289
#53 0x0000004001f4f726 in invokeCompileMethodHelper (jitMgr=0x40000b1220, comp=0x4001814528, info=0x4001814380, jitFlags=..., nativeEntry=0x4001814ed0, nativeSizeOfCode=0x400181435c) at /runtime/src/coreclr/vm/jitinterface.cpp:12520
#54 0x0000004001f4f894 in invokeCompileMethod (jitMgr=0x400180af28, comp=0xffffffff80131522, info=0xa6af15a049a54a00, jitFlags=..., nativeEntry=0x4001bfc690, nativeSizeOfCode=0x800) at /runtime/src/coreclr/vm/jitinterface.cpp:12583
#55 0x0000004001f5070e in UnsafeJitFunction (config=<optimized out>, ILHeader=<optimized out>, pJitFlags=0x4001814fe0, pSizeOfCode=0x40018150bc) at /runtime/src/coreclr/vm/jitinterface.cpp:13028
#56 0x0000004001f97b4c in MethodDesc::JitCompileCodeLocked (this=0x3f840de6b8, pConfig=0x40018153f0, pilHeader=0x4001815080, pEntry=0x608c0260e0, pSizeOfCode=0x40018150bc) at /runtime/src/coreclr/vm/prestub.cpp:939
#57 0x0000004001f9742a in MethodDesc::JitCompileCodeLockedEventWrapper (this=0x3f840de6b8, pConfig=0x40018153f0, pEntry=0x608c0260e0) at /runtime/src/coreclr/vm/prestub.cpp:820
#58 0x0000004001f96ae4 in MethodDesc::JitCompileCode (this=0x3f840de6b8, pConfig=0x40018153f0) at /runtime/src/coreclr/vm/prestub.cpp:707
#59 0x0000004001f96226 in MethodDesc::PrepareILBasedCode (this=0x3f840de6b8, pConfig=0x40018153f0) at /runtime/src/coreclr/vm/prestub.cpp:433
#60 0x0000004001ee02c2 in CodeVersionManager::PublishVersionableCodeIfNecessary (this=0x40000ad2d0, pMethodDesc=0x3f840de6b8, callerGCMode=CallerGCMode::Coop, doBackpatchRef=0x40018154e8, doFullBackpatchRef=<optimized out>) at /runtime/src/coreclr/vm/codeversion.cpp:1734
#61 0x0000004001f9b830 in MethodDesc::DoPrestub (this=0x3f840de6b8, pDispatchingMT=0x3f840de738, callerGCMode=CallerGCMode::Coop) at /runtime/src/coreclr/vm/prestub.cpp:2595
#62 0x0000004001f9afee in PreStubWorker (pTransitionBlock=<optimized out>, pMD=0x3f840de6b8) at /runtime/src/coreclr/vm/prestub.cpp:2424
#63 0x000000400221a49a in ThePreStub () at /runtime/src/coreclr/vm/riscv64/asmhelpers.S:378
#64 0x0000003f83f08c68 in ?? ()
```
</details>